### PR TITLE
Bump k8s to 1.28

### PR DESCRIPTION
**Pull request by tklauser**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,10 +8,6 @@ replace (
 
 	go.universe.tf/metallb => github.com/cilium/metallb v0.1.1-0.20220829170633-5d7dfb1129f7
 
-	// Using fork of client-go.
-	// Contains fix for exponential backoff for informers.
-	k8s.io/client-go => github.com/cilium/client-go v0.27.2-fix
```
Comment (rolinh) [inline go.mod:13]: :heart:

**Review by rolinh — APPROVED** [review]

File: connectivity/check/policy.go [hunk 2]
```diff
@@ -146,7 +146,6 @@ func createOrUpdateKNP(ctx context.Context, client *k8s.Client, knp *networkingv
 	// Overload the field that should stay unchanged.
 	policy.ObjectMeta.Labels = knp.ObjectMeta.Labels
 	policy.Spec = knp.Spec
-	policy.Status = networkingv1.NetworkPolicyStatus{}
```
Comment (michi-covalent) [inline connectivity/check/policy.go:149]: what happened to this field

**Review by michi-covalent — APPROVED** [review]

File: connectivity/check/policy.go (hunk 2 above)
Comment (tklauser) [reply connectivity/check/policy.go:149]: Looks like it was removed upstream by https://github.com/kubernetes/kubernetes/pull/115843 ref. https://github.com/kubernetes/enhancements/issues/2943

Comment (michi-covalent) [reply connectivity/check/policy.go:149]: how disappointing 😿

**Pull request merged**